pkgs/applications/editors/vscode/vscode-oss.nix

```diff
@@ -0,0 +1,331 @@
+    mkdir -p ${path}/node_modules/esbuild/bin
+    jq "del(.scripts.postinstall)" ${path}/node_modules/esbuild/package.json | sponge ${path}/node_modules/esbuild/package.json
+    sed -i 's/${version}/${esbuild'.version}/g' ${path}/node_modules/esbuild/lib/main.js
+    ln -s -f ${esbuild'}/bin/esbuild ${path}/node_modules/esbuild/bin/esbuild
+  '';
+in
+
+stdenv.mkDerivation (finalAttrs: {
+  pname = executableName;
+  version = "1.89.1";
+  commit = "dc96b837cf6bb4af9cd736aa3af08cf8279f7685";
+
+  src = fetchFromGitHub {
+    owner = "microsoft";
+    repo = "vscode";
+    rev = finalAttrs.version;
+    sha256 = "sha256-z4VA1pv+RPAzUOH/yK6EB84h4DOFG5TcRH443N7EIL0=";
+  };
+
+  passthru.product =
+    # Base
+    product
+    // {
+      inherit (finalAttrs) version;
+      inherit extensionsGallery;
+
+      nameShort = shortName;
+      nameLong = longName;
+      applicationName = executableName;
+    }
+    # Extra overrides
+    // productOverrides;
+
+  yarnCache = stdenv.mkDerivation (
+    let
+      inherit (finalAttrs) pname version src;
+    in
+    {
+      inherit src;
+      name = "${pname}-${version}-yarn-cache";
+      GIT_SSL_CAINFO = "${cacert}/etc/ssl/certs/ca-bundle.crt";
+      NODE_EXTRA_CA_CERTS = "${cacert}/etc/ssl/certs/ca-bundle.crt";
+      nativeBuildInputs = [ yarn ];
+      buildPhase = ''
+        export HOME=$PWD
+        yarn config set yarn-offline-mirror $out
+        find "$PWD" -name "yarn.lock" -printf "%h\n" | \
+          xargs -I {}                                                          \
+            yarn --cwd {}                                                      \
+            --frozen-lockfile                                                  \
+            --ignore-scripts                                                   \
+            --ignore-platform                                                  \
+            --ignore-engines                                                   \
+            --no-progress                                                      \
+            --non-interactive
+      '';
+
+      outputHashMode = "recursive";
+      outputHashAlgo = "sha256";
+      outputHash = "sha256-7Qy0xMLcvmZ43EcNbsy7lko0nsXlbVYSMiq6aa4LuoQ=";
+    }
+  );
+
+  outputs = [
+    # Desktop version.
+    "out"
+
+    # Remote extension host, for vscode remote development
+    "reh"
+
+    # Web extension host, run extension host in CLI, client in browser
+    "rehweb"
+  ];
+
+  nativeBuildInputs =
+    [
+      yarn'
+      python3
+      nodejs
+      jq
+      moreutils
+      pkg-config
+      makeWrapper
+    ]
+    ++ lib.optionals stdenv.isDarwin [ darwin.cctools ]
+    ++ lib.optionals (stdenv.isDarwin && stdenv.isAarch64) [ autoSignDarwinBinariesHook ];
+
+  buildInputs = [
+    krb5
+    libsecret
+    xorg.libX11
+    xorg.libxkbfile
+  ] ++ lib.optionals stdenv.isDarwin [ darwin.apple_sdk.frameworks.Cocoa ];
+
+  patches = map (name: ./vscode-oss/patches/${name}) (
```

Comment:
I think we might need to use filterSource here to not copy the entire tree again. There should be an issue about this.